test/protocol/ProtocolInitializationHandlerTest.js

```diff
@@ -328,4 +343,163 @@ describe("ProtocolInitializationHandler", async function () {
+      version = ethers.utils.formatBytes32String("2.2.0");
+
+      // Prepare cut data, and attach correct address to configHandler
+      facetCut = getFacetAddCut(configHandler);
+      configHandler = configHandler.attach(protocolDiamond.address);
+    });
+
+    it("Should emit MaxPremintedVouchersChanged event", async function () {
+      const calldataProtocolInitialization = deployedProtocolInitializationFacet.contract.interface.encodeFunctionData(
+        "initialize",
+        [version, [], [], true, initializationData, [], []]
+      );
+
+      // Make the cut, check the event
+      await expect(
+        diamondCutFacet.diamondCut(
+          [facetCut],
+          deployedProtocolInitializationFacet.contract.address,
+          calldataProtocolInitialization,
+          await getFees(maxPriorityFeePerGas)
+        )
+      )
+        .to.emit(configHandler, "MaxPremintedVouchersChanged")
+        .withArgs(maxPremintedVouchers, deployer.address);
+    });
+
+    it("Should update state", async function () {
+      const calldataProtocolInitialization = deployedProtocolInitializationFacet.contract.interface.encodeFunctionData(
+        "initialize",
+        [version, [], [], true, initializationData, [], []]
+      );
+
+      // Make the cut, check the event
+      await diamondCutFacet.diamondCut(
+        [facetCut],
+        deployedProtocolInitializationFacet.contract.address,
+        calldataProtocolInitialization,
+        await getFees(maxPriorityFeePerGas)
+      );
+
+      // Verify that new value is stored
+      expect(await configHandler.connect(rando).getMaxPremintedVouchers()).to.equal(maxPremintedVouchers);
+    });
+
+    context("💔 Revert Reasons", async function () {
+      it("Max preminted vouchers is zero", async function () {
+        // set invalid maxPremintedVouchers
+        maxPremintedVouchers = "0";
+        initializationData = ethers.utils.defaultAbiCoder.encode(["uint256"], [maxPremintedVouchers]);
+
+        const calldataProtocolInitialization =
+          deployedProtocolInitializationFacet.contract.interface.encodeFunctionData("initialize", [
+            version,
+            [],
+            [],
+            true,
+            initializationData,
+            [],
+            [],
+          ]);
+
+        // make diamond cut, expect revert
+        await expect(
+          diamondCutFacet.diamondCut(
+            [facetCut],
+            deployedProtocolInitializationFacet.contract.address,
+            calldataProtocolInitialization,
+            await getFees(maxPriorityFeePerGas)
+          )
+        ).to.be.revertedWith(RevertReasons.VALUE_ZERO_NOT_ALLOWED);
+      });
+
+      it("Current version is not 0", async () => {
+        // Deploy higher version
+        version = "2.3.0";
+        const interfaceId = InterfaceIds[interfaceImplementers["ProtocolInitializationFacet"]];
+        const { deployedFacets } = await deployAndCutFacets(
```

Comment:
```suggestion
               const {
          deployedFacets: [{ contract: deployedProtocolInitializationFacet }],
        } = await deployAndCutFacets(
```
just being boring again

Reply:
changed.